packages/rule-tester/src/RuleTester.ts

```diff
@@ -197,10 +197,7 @@ export class RuleTester extends TestFramework {
         ? this.#testerConfig.defaultFilenames.tsx
         : this.#testerConfig.defaultFilenames.ts;
       if (resolvedOptions.project) {
-        return path.join(
-          resolvedOptions.tsconfigRootDir ?? process.cwd(),
-          filename,
-        );
+        return path.resolve(process.cwd(), filename);
```

Comment:
Continuing the [suggested fix](https://github.com/typescript-eslint/typescript-eslint/issues/8271#issuecomment-1899534142):

> The fix here probably should be to make `rule-tester` absolutise the path so it matches ESLint.

@c0sta have you tried that?